src/tests/ftest/cart/util/daos_sys_logscan.py

```diff
@@ -0,0 +1,553 @@
+            term = self._cur_term[puuid]
+            if term < 1:
+                self._warn(f"pool {puuid} I don't know what term it is ({term})!", fname, line)
+                return True
+            if ver not in self._pools[puuid][term]["maps"]:
+                self._warn(f"pool {puuid} term {term} ver {ver} not in maps - add placeholder",
+                           fname, line)
+                self._pools[puuid][term]["maps"][ver] = {
+                    "carryover": False,
+                    "from_ver": ver,
+                    "time": "xx/xx-xx:xx:xx.xx",
+                    "rb_gens": {}
+                }
+                self._pools[puuid][term]["maps"][ver]["rb_gens"][gen] = {
+                    "op": op,
+                    "start_time": "xx/xx-xx:xx:xx.xx",
+                    "time": "xx/xx-xx:xx:xx.xx",
+                    "started": True,
+                    "scanning": False,
+                    "pulling": False,
+                    "completed": False,
+                    "aborted": False,
+                    "duration": 0
+                }
+            if gen in self._pools[puuid][term]["maps"][ver]["rb_gens"]:
+                existing_op = self._pools[puuid][term]["maps"][ver]["rb_gens"][gen]["op"]
+                if op != existing_op:
+                    self._warn(f"rb={puuid}/{ver}/{gen}/{existing_op} != line op {op}", fname, line)
+            if status == "scanning":
+                self._pools[puuid][term]["maps"][ver]["rb_gens"][gen]["scanning"] = True
+            elif status == "pulling":
+                self._pools[puuid][term]["maps"][ver]["rb_gens"][gen]["pulling"] = True
+            elif status == "completed":
+                self._pools[puuid][term]["maps"][ver]["rb_gens"][gen]["completed"] = True
+            elif status == "aborted":
+                self._pools[puuid][term]["maps"][ver]["rb_gens"][gen]["aborted"] = True
+            self._pools[puuid][term]["maps"][ver]["rb_gens"][gen]["time"] = datetime
+            self._pools[puuid][term]["maps"][ver]["rb_gens"][gen]["duration"] = dur
+            if self._debug:
+                print(f"{datetime} FOUND rebuild UPDATE term={term} rb={puuid}/{ver}/{gen}/{op} "
+                      f"STATUS={status}, DUR={dur} seconds rank {rank}\t{host}\tPID {pid}\t{fname}")
+            return True
+        return False
+
+    def _match_legacy_ps_rb_status(self, fname, line, pid, rank):
+        # Do not match on legacy rebuild log format if we found new format
+        if not self._check_rb_legacy_fmt:
+            return False
+        msg, host, datetime = self._get_line_components(line)
+        match = self.re_old_ldr_status.match(msg)
+        if match:
+            # Disable checking for new rebuild log format, to save execution time
+            self._check_rb_new_fmt = False
+            op = match.group(1)
+            status = match.group(2)
+            puuid = match.group(3)
+            log_ldr = int(match.group(4))
```

Comment:
Nit, could be replaced with:
`op, status, puuid, log_ldr = self._get_rb_components(match)`